hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/scm/protocolPB/StorageContainerLocationProtocolClientSideTranslatorPB.java

```diff
@@ -758,6 +758,7 @@ public ScmInfo getScmInfo() throws IOException {
     ScmInfo.Builder builder = new ScmInfo.Builder()
         .setClusterId(resp.getClusterId())
         .setScmId(resp.getScmId())
+        .setScmRatisEnabled(resp.getScmRatisEnabled())
```

Comment:
Here to be backward compatible, we should first check whether the response has "scmRatisEnabled" field. If it doesn't have, which means the SCM server is an old version server, then we need to set this ScmRatisEnabled field in other way, for example, the size of RolesList.

Reply:
> Here to be backward compatible, we should first check whether the response has "scmRatisEnabled" field. 

Thank you for the reminder! It's indeed very necessary. I will check if `ScmRatisEnabled` exists.

> If it doesn't have, which means the SCM server is an old version server, then we need to set this ScmRatisEnabled field in other way, for example, the size of RolesList.

We use the size of `RolesList` to determine if Ratis is enabled. If RolesList.size > 1, we consider Ratis enabled. If RolesList.size <= 1, we consider Ratis not enabled ？

Reply:
Although there might be one corner case not covered, it should be OK to check the roleList size.

Reply:
Thank you for your suggestion! I will continue to improve the code.

Reply:
@ChenSammi Could you review this PR again? Thank you very much!